scripts/west_commands/build.py

```diff
@@ -22,7 +22,7 @@
 BUILD_USAGE = '''\
 west build [-h] [-b BOARD] [-d BUILD_DIR]
            [-t TARGET] [-p {auto, always, never}] [-c] [--cmake-only]
-           [-n] [-o BUILD_OPT] [-f]
+           [-n] [-o BUILD_OPT] [-f] [-s SNIPPET]
```

Comment:
> * Although -s has been deprecated for some time, then I think it's good practice to have a period between removing the argument before using that same argument for something new.

In an ideal world yes, but the end result is going to be the same. If `-s` is removed, any scripts using it will crash. If `-s` is reused, any scripts using it will crash. Unfortunately I can't go back and remove it a year ago. Of course if the name changes from "snippet" then there is no need to use `-s`.

> * Why should snippet have it's own west build argument when CONF_FILE, DTC_OVERLAY_FILE, SHIELD etc. doesn't have?

Because its goal is to cover common use cases, which IMO means the ease-of-use of `west build -b board -s usb samples/basic/minimal` over `west build -b board samples/basic/minimal -- -DSNIPPETS="usb"` warrants it.

Reply:
I would say `-DOVERLAY_CONFIG` or `DTC_OVERLAY_FILE` are much better candidates for having extra arguments than snippets.

Reply:
I disagree; I think snippets do rise to the level of deserving their own option.

I want to ask what the practical differences between snippets and shields are at this point, in terms of their behavior. Shields seem to be a special case of snippets, morally speaking.

Taking @petejohanson since he's taken an interest in adding more comprehensive shield related options to west build elsewhere.